models/analytics/mgramseva/demand_collection.sql

```diff
@@ -24,97 +24,77 @@
 {{ config(materialized='table') }}
 
 WITH table_p AS (
-    SELECT
-        consumercode,
+    SELECT 
+        consumercode, 
         tenantid,
-        TO_CHAR(paymentdate, 'YYYY-MM-DD') AS meeting_date,
-        TO_CHAR(paymentdate, 'Month') AS reporting_month,
+        TO_CHAR(paymentdate, 'YYYY-MM-DD') AS meeting_date, 
+        TO_CHAR(paymentdate, 'Month') AS reporting_month, 
+        EXTRACT(YEAR FROM paymentdate) AS reporting_year,
         SUM(totalpaymentpaid) AS amount_p
-    FROM {{ ref('paymentdetails') }}
-    GROUP BY
-        consumercode,
-        tenantid,
-        TO_CHAR(paymentdate, 'YYYY-MM-DD'),
-        TO_CHAR(paymentdate, 'Month')
-    ORDER BY consumercode, TO_CHAR(paymentdate, 'YYYY-MM-DD')
+    FROM {{ref('paymentdetails')}}
+    GROUP BY consumercode, tenantid, TO_CHAR(paymentdate, 'Month'), EXTRACT(YEAR FROM paymentdate), TO_CHAR(paymentdate, 'YYYY-MM-DD')
 ),
 
 table_d AS (
-    SELECT
-        consumercode,
+    SELECT 
+        consumercode, 
         tenantid,
-        TO_CHAR(demandtodate, 'YYYY-MM-DD') AS meeting_date,
-        TO_CHAR(demandtodate, 'Month') AS reporting_month,
+        TO_CHAR(demandtodate, 'YYYY-MM-DD') AS meeting_date, 
+        TO_CHAR(demandtodate, 'Month') AS reporting_month, 
+        EXTRACT(YEAR FROM demandtodate) AS reporting_year,
         SUM(demandamount) AS amount_d
-    FROM {{ ref('demanddetails') }}
-    GROUP BY
-        consumercode,
-        tenantid,
-        TO_CHAR(demandtodate, 'YYYY-MM-DD'),
-        TO_CHAR(demandtodate, 'Month')
-    ORDER BY consumercode, TO_CHAR(demandtodate, 'YYYY-MM-DD')
+    FROM {{ref('demanddetails')}}
+    GROUP BY consumercode, tenantid, TO_CHAR(demandtodate, 'Month'), EXTRACT(YEAR FROM demandtodate), TO_CHAR(demandtodate, 'YYYY-MM-DD')
 ),
 
 water_connections AS (
-    SELECT
-        TO_TIMESTAMP(
-            COALESCE(table_d.meeting_date, table_p.meeting_date), 'YYYY-MM-DD'
-        )::date AS meeting_date,
-        COALESCE(table_d.consumercode, table_p.consumercode) AS consumercode,
-        -- Convert timestampz to date
+    SELECT 
+        COALESCE(table_d.consumercode, table_p.consumercode) AS consumercode, 
         COALESCE(table_d.tenantid, table_p.tenantid) AS tenantid,
-        COALESCE(
```

Comment:
@nishika26 are we going to write any test cases for the models ?

Reply:
dbt test cases are already written and updated for all the dbt models